### PR TITLE
Add maximum record size config

### PR DESCRIPTION
**Pull request by foadnh**

### CODE REVIEW

File: aggregate/aggregator.go [hunk 1]
```diff
@@ -19,31 +19,46 @@ var (
...
+	maximumRecordSize int
+	maxAggRecordSize  int
+	stringGen         *util.RandomStringGenerator
 }
 
-// NewAggregator create a new aggregator
-func NewAggregator(stringGen *util.RandomStringGenerator) *Aggregator {
+// Config is for aggregation customizations.
+type Config struct {
+	MaximumRecordSize *int
```
Comment (chengchengmu) [inline aggregate/aggregator.go:40]: OOC is the convention different for this kind of struct ? other variables are in camelCase

File: aggregate/aggregator.go [hunk 2]
```diff
@@ -91,7 +106,7 @@ func (a *Aggregator) AddRecord(partitionKey string, hasPartitionKey bool, data [
 	pkeyFieldSize := protowire.SizeVarint(pKeyIdx) + fieldNumberSize
 	// Total size is byte size of data + pkey field + field number of parent proto
 
-	if a.getSize()+protowire.SizeBytes(dataFieldSize+pkeyFieldSize)+fieldNumberSize+pKeyAddedSize >= maximumRecordSize {
+	if a.getSize()+protowire.SizeBytes(dataFieldSize+pkeyFieldSize)+fieldNumberSize+pKeyAddedSize >= defaultMaximumRecordSize {
```
Comment (chengchengmu) [inline aggregate/aggregator.go]: should it be using the `maximumRecordSize` config ?
```suggestion
	if a.getSize()+protowire.SizeBytes(dataFieldSize+pkeyFieldSize)+fieldNumberSize+pKeyAddedSize >= a.maximumRecordSize {
```

File: aggregate/aggregator.go [hunk 3]
```diff
@@ -19,31 +19,46 @@ var (
...
+	a := &Aggregator{
 		partitionKeys:    make(map[string]uint64, 0),
 		records:          make([]*Record, 0),
 		maxAggRecordSize: defaultMaxAggRecordSize,
 		aggSize:          initialAggRecordSize,
 		stringGen:        stringGen,
 	}
+
+	if cfg.MaximumRecordSize != nil {
+		a.maximumRecordSize = *cfg.MaximumRecordSize
```
Comment (chengchengmu) [inline aggregate/aggregator.go:56]: do we need to default `a.maximumRecordSize` to `defaultMaximumRecordSize` when `cfg.MaximumRecordSize` is nil ?

**Review by chengchengmu — COMMENTED** [review]
Really good to have those parameters configurable !

Left some comments, let me know if they make sense or not :) happy to chat on slack/zoom

File: aggregate/aggregator.go (hunk 1 above)
Comment (foadnh) [reply aggregate/aggregator.go:40]: Do you mean starting with a capital character? That is to expose variables to other packages to be able to set it. `kinesis` package sets it.

File: aggregate/aggregator.go (hunk 2 above)
Comment (foadnh) [reply aggregate/aggregator.go]: Oh right. My bad, I missed it.

File: aggregate/aggregator.go (hunk 3 above)
Comment (foadnh) [reply aggregate/aggregator.go:56]: Ah, I missed it. Thanks, fixed.

**Review by chengchengmu — APPROVED** [review]
LGTM !

**Pull request merged**